### PR TITLE
Enable Hill Slope RWD

### PR DESCRIPTION
**Pull request by mmcfarland**
Hill slope is a rapid water delineation function which does not use snapping.  Allow Hill Slope to pass a snapping option through to the RWD service.

##### Testing
Run a `debugcelery` session and delineate a watershed with both the Stream Network and Hill Slope functions.  Hill Slope should output a debug logging statement with the querystring `?snapping=false`

### CODE REVIEW

**Comment by mmcfarland** [comment]
Results for Hill Slope are not returning correctly due to https://github.com/WikiWatershed/rapid-watershed-delineation/issues/14

**Comment by lewfish** [comment]
+1. I'm now re-provisioning my VM to test.

**Comment by lewfish** [comment]
> Results for Hill Slope are not returning correctly due to WikiWatershed/rapid-watershed-delineation#14

Everything works except for this issue. I can test this again after the issue is fixed if you want.

**Comment by mmcfarland** [comment]
Added a commit that pulls the latest `docker-rwd` which has the code fix for Hill Slope.  To test, you'll need to `vagrant provision worker` and try delineating a Hill Slope again.

File: src/mmw/apps/modeling/views.py [hunk 1]
```diff
@@ -164,10 +164,15 @@ def start_rwd(request, format=None):
     user = request.user if request.user.is_authenticated() else None
     created = now()
     location = request.POST['location']
+
+    # Parse out the JS style T/F to a boolean
+    snappingParam = request.POST['snappingOn']
```
Comment (kdeloach) [inline src/mmw/apps/modeling/views.py]: Consider using `request.POST.get('snappingOn', False)` to prevent the possibility of a `KeyError`.

**Comment by kdeloach** [comment]
+1

**Comment by lewfish** [comment]
Hill slope seems to work, but for some reason the analyze view is failing which is surprising since the area is so small. I get back the error 

```
HTTPConnectionPool(host='localhost', port=8090): Max retries exceeded with url: /jobs?context=geoprocessing&appName=geoprocessing-0.4.0&classPath=org.wikiwatershed.mmw.geoprocessing.SummaryJob (Caused by NewConnectionError('<requests.packages.urllib3.connection.HTTPConnection object at 0x7fc493ca9410>: Failed to establish a new connection: [Errno 111] Connection refused'
```

screen shot 2016-01-25 at 11 32 41 am 2

**Comment by mmcfarland** [comment]
Looks like this is an error with creating very small (presumably sub-30m) AoIs, which this function occasionally does.  There's a division by 0 error in those cases.  Generally speaking, delineating a flat area is more prone to making tiny AoIs where "hilly" areas tend to produce merely small ones.  I might end up making a new card to handle small AoIs as a separate issue.


```
ZeroDivisionError('float division by zero',)
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "apps/modeling/tasks.py", line 106, in histogram_to_survey
    results = data_to_survey(data)
  File "/usr/local/lib/python2.7/dist-packages/statsd/client.py", line 26, in wrapper
    return f(*args, **kw)
  File "apps/modeling/geoprocessing.py", line 256, in data_to_survey
    return histogram_to_x(data, nucleus, update_rule, after_rule)
  File "apps/modeling/geoprocessing.py", line 148, in histogram_to_x
    after_rule(total, retval)
  File "apps/modeling/geoprocessing.py", line 240, in after_rule
    survey[0]['categories'] = [update_pcts(v, count) for k, v in land]
  File "apps/modeling/geoprocessing.py", line 202, in update_pcts
    entry['coverage'] = float(area) / count
ZeroDivisionError: float division by zero
```

**Comment by mmcfarland** [comment]
Added 09f19d7 to prevent division by 0.  There are still no results for such small AoIs (all zeros), but we don't have anything to report for them anyway.  We'll ultimately need a better method to inform users that the AoI is too small, and maybe a separate way to handle small auto-generated AoIs like Hill Slope (unless there's a bug in RWD responsible for their size).

**Comment by mmcfarland** [comment]
Hill Slope also seems to occasionally not produce any output, and triggers this exception:

https://github.com/WikiWatershed/rapid-watershed-delineation/blob/develop/src/api/main.py#L73

We'll have to bring that up with the TauDEM team.

**Comment by lewfish** [comment]
+1

**Pull request merged**